### PR TITLE
refactor: modularize tasklist-data-generator module

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description
Same as https://github.com/camunda/camunda/pull/18188

## Related issues

relates to https://github.com/camunda/camunda/issues/18535

### CODE REVIEW

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/api/rest/v1/controllers/internal/DevUtilExternalController.java [hunk 1]
```diff
@@ -84,7 +81,6 @@ public ResponseEntity<?> recreateData() throws IOException {
     processCache.clearCache();
     schemaManager.createSchema();
     searchEngineUserDetailsService.initializeUsers();
-    devDataGenerator.createDemoUsers();
```
Comment (houssain-barouni) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/api/rest/v1/controllers/internal/DevUtilExternalController.java:87]: `DO_NOTHING` DataGenerator bean was autowired here, so it is not required

File: tasklist/common/src/main/java/io/camunda/tasklist/data/DataGenerator.java [hunk 2]
```diff
@@ -9,34 +9,11 @@
 
 public interface DataGenerator {
 
-  DataGenerator DO_NOTHING =
```
Comment (Zelldon) [inline tasklist/common/src/main/java/io/camunda/tasklist/data/DataGenerator.java:12]: ❓ This is not needed anymore?

File: tasklist/data-generator/src/main/java/io/camunda/tasklist/data/DevDataGeneratorAbstract.java [hunk 3]
```diff
@@ -28,9 +29,11 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.DependsOn;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
+@DependsOn("schemaStartup")
```
Comment (Zelldon) [inline tasklist/data-generator/src/main/java/io/camunda/tasklist/data/DevDataGeneratorAbstract.java:36]: It somehow feels so weird to depend on something which is not in the same module, like here we now depend on something from the web-app. So I can't use the data generator standalone ? 🤔 But Idk I guess I'm not familiar enough with spring

**Review by ChrisKujawa — APPROVED** [review]
I'm always not sure whether I'm the right person to review such changes :D Spring magic feels always too hard to review, as there is always some context I feel to be missed (all the other beans lying around, etc.) 

What I understood from the PR is:

You move the data-generator dependency from web-app to dist? 
You change it in a way that the data generation is enabled only when dev-data profile is given. I think this is also the reason why we no longer need to Noop data generator, right?

The data generator depends on the web-app schema startup, after it is finished the data generator will start

So yes I guess LGTM, but I hope at some point we clean up the mess of using spring beans/components everywhere, as the dependency tree/bootstrap is totally unclear, especially in reviews.

**Comment by houssain-barouni** [comment]
> I'm always not sure whether I'm the right person to review such changes :D Spring magic feels always too hard to review, as there is always some context I feel to be missed (all the other beans lying around, etc.)
> 
> What I understood from the PR is:
> 
> You move the data-generator dependency from web-app to dist? You change it in a way that the data generation is enabled only when dev-data profile is given. I think this is also the reason why we no longer need to Noop data generator, right?
> 
> The data generator depends on the web-app schema startup, after it is finished the data generator will start
> 
> So yes I guess LGTM, but I hope at some point we clean up the mess of using spring beans/components everywhere, as the dependency tree/bootstrap is totally unclear, especially in reviews.

Your understanding is correct, I needed these changes for a different pull request https://github.com/camunda/camunda/pull/18604, where I needed to get rid of the to Noop data generator usage in `StandaloneTasklist`. I could simply have moved it to the newly created `TasklistModuleConfiguration`, but I preferred to do the exactly the same as done for Operate

**Review by nathansandi — APPROVED** [review]
💯

**Pull request merged**